docs/roles.md

```diff
@@ -0,0 +1,151 @@
+- Lookup Party Information - `GET /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Retrieve Transaction Request Information - `GET /transactionRequests/{ID}`
+- Perform Transaction Request - `POST /transactionRequests`
+- Retrieve Quote Information - `GET /quotes/{ID}`
+- Calculate Quote - `POST /quotes`
+- Retrieve Bulk Quote Information - `GET /bulkQuotes/{ID}`
+- Calculate Bulk Quote - `POST /bulkQuotes`
+- Perform Authorization - `GET /authorizations/{ID}`
+- Retrieve Transfer Information - `GET /transfers/{ID}`
+- Perform Transfer - `POST /transfers`
+- Retrieve Bulk Transfer Information - `GET /bulkTransfers/{ID}`
+- Perform Bulk Transfer - `POST /bulkTransfers`
+- Retrieve Transaction Information - `GET /transactions/{ID}`
+
+> New PISP Calls
+- Create Consent Request - `POST /consentRequests/{ID}`
+- Update Consent Request - `PUT /consentRequests/{ID}`
+- Perform Authorization on this Thing `POST /authorizations/{ID}` __newly added _verb_ for PISP case__
+- Update Consent - `POST /consents/{ID}`
+- Initiate 3rd party request - `POST /thirdPartyRequest/transaction/{ID}`
+- Get 3rd party request information - `GET /thirdPartyRequest/transaction/{ID}`
+
+
+### Inbound (Callback from Switch -> Participant)
+
+- Return Bulk Participant Information - `PUT /participants/{ID}`
+- Return Bulk Participant Information Error - `PUT /participants/{ID}/error`
+- Return Participant Information - `PUT /participants/{Type}/{ID}, /participants/{Type}/{ID}/{SubId}`
+- Return Participant Information Error - `PUT /participants/{Type}/{ID}/error, /participants/{Type}/{ID}/{SubId}/error`
+- Return Party Information - `PUT /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Return Party Information Error - `PUT /parties/{Type}/{ID}/error, /parties/{Type}/{ID}/{SubId}/error`
+- Return Transaction Request Information - `PUT /transactionRequests/{ID}`
+- Return Transaction Request Information Error - `PUT /transactionRequests/{ID}/error`
+- Return Quote Information - `PUT /quotes/{ID}`
+- Return Quote Information Error - `PUT /quotes/{ID}/error`
+- Return Bulk Quote Information - `PUT /bulkQuotes/{ID}`
+- Return Bulk Quote Information Error - `PUT /bulkQuotes/{ID}/error`
+- Return Authorization Result - `PUT /authorizations/{ID}`
+- Return Authorization Error - `PUT /authorizations/{ID}/error`
+- Return Transfer Information - `PUT /transfers/{ID}`
+- Return Transfer Information Error - `PUT /transfers/{ID}/error`
+- Return Bulk Transfer Information - `PUT /bulkTransfers/{ID}`
+- Return Bulk Transfer Information Error - `PUT /bulkTransfers/{ID}/error`
+- Return Transaction Information - `PUT /transactions/{ID}`
+- Return Transaction Information Error - `PUT /transactions/{ID}/error`
+
+> New PISP Calls
+- Update Consent Request - `PUT /consentRequests/{ID`
+- Update Consent Request Error - `PUT /consentRequests/{ID}/error`
+- Update Consent - `POST /consents/{ID}`
```

Comment:
if this is `Inbound`, shouldn't be `PUT` instead of `POST` for `/consents/{ID}`?

Reply:
I think that's right: it looks like a repeat of l.47, and there is no substantive response required to this call. As I understand it, it's simply informative; however, we might still require a PUT to confirm to the DFSP that the PISP had registered the permission. Lewis, what do you think?

Reply:
Yes I think your're right @eoln.

My original thought here was that the `consent` is created by the DFSP as the result of a successful `consentRequest` (that's the call on L:47, which I have renamed **Create Consent**) and that the PISP has no notion of that created `consent` until it it recieves a `POST`.

If, however, we are happy with `PUT` to act as a Update or Create (which I think we should be, since that is how HTTP works), then this should be:
```
- Create Or Update Consent - `PUT /consents/{ID}`
```